pkg/daemon/ceph/osd/encryption.go

```diff
@@ -172,6 +173,17 @@ func openEncryptedDevice(context *clusterd.Context, disk, target, passphrase str
 	return nil
 }
 
+func removeEncryptedDevice(context *clusterd.Context, target string) error {
+	args := []string{"remove", "--force", target}
+	output, err := context.Executor.ExecuteCommandWithTimeout(removeEncryptedDeviceCmdTimeOut, "dmsetup", args...)
+	// ignore error if no device was found.
+	if err != nil && strings.Contains(output, "No such device") {
```

Comment:
Let's check for `ENODEV` instead of the strings.Contains

Reply:
Also, don't we want to ignore it if the device does not exist?

Reply:
> Also, don't we want to ignore it if the device does not exist?

forgot `!` :facepalm: , fixed this 

>It's a bit of "weaker" test than what the OSD code does but it could be sufficient.

Considering the previous part of the code that leads here detected the encrypted device not to be open,
I figured it would be simpler to just remove the device and open it again, skipping all the checks in between.

> Let's check for `ENODEV` instead of the strings.Contains

Is there any example code to do that 
Current error thrown is 
(i ran the cmd twice to get this)
```
output: device-mapper: remove ioctl on ocs-deviceset-thin-csi-1-data-0cg76k-block-dmcrypt  failed: No such device or address
error: Command failed.exit status 1
```